patches/tModLoader/Terraria/ModLoader/ModNPC.cs

```diff
@@ -158,10 +158,9 @@ public abstract class ModNPC : ModTexturedType
 
 		/// <summary>
 		/// Allows you to set the town NPC profile that this NPC uses.
+		/// By default, returns null, meaning that the NPC doesn't use one.
 		/// </summary>
-		/// <param name="database">The list of town NPC profiles that currently exist.</param>
-		public virtual void SetTownNPCProfile(Dictionary<int, ITownNPCProfile> database) {
-		}
+		public virtual ITownNPCProfile TownNPCProfile => null;
```

Comment:
Is there a reason this is a property and not a method? Clashes with `GetTextureNPCShouldUse` which is a method

Reply:
originally, I'd figured it'd be more convenient for it to be a property, but now that I look at it, havin' it be a method will probably be more convenient in the event somebody wants to do something like use different profiles for an NPC dependin' on progression. also just helps keep things consistent